fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -458,8 +475,417 @@ Value makePadding(int size) {
+		                                     BlobCipherMetrics::BACKUP);
+		Arena arena;
+		int64_t payloadSize = wPtr - dataPayloadStart;
+		auto encryptedData = encryptor.encrypt(dataPayloadStart, payloadSize, encryptHeader, arena);
+
+		// re-write encrypted data to buffer
+		std::memcpy(dataPayloadStart, encryptedData->begin(), payloadSize);
+	}
+
+	ACTOR static Future<Void> updateEncryptionKeys(EncryptedRangeFileWriter* self,
+	                                               KeyRef key,
+	                                               Reference<TenantEntryCache<Void>> cache) {
+		state std::pair<int64_t, TenantName> curTenantInfo = wait(getTenantIdName(key, cache));
+		state Reference<AsyncVar<ClientDBInfo> const> dbInfo = self->cx->clientInfo;
+		// Get and Set Header Cipher Key
+		TextAndHeaderCipherKeys systemCipherKeys =
+		    wait(getLatestSystemEncryptCipherKeys(dbInfo, BlobCipherMetrics::BACKUP));
+		self->cipherKeys.headerCipherKey = systemCipherKeys.cipherHeaderKey;
+
+		// Get text cipher key
+		std::unordered_map<EncryptCipherDomainId, EncryptCipherDomainName> domains;
+		domains.emplace(curTenantInfo.first, curTenantInfo.second);
+		// Get the latest write encryption key for the given tenant
+		std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>> textCipherKeys =
+		    wait(getLatestEncryptCipherKeys(dbInfo, domains, BlobCipherMetrics::BACKUP));
+		ASSERT(textCipherKeys.find(curTenantInfo.first) != textCipherKeys.end());
+		self->cipherKeys.textCipherKey = textCipherKeys.at(curTenantInfo.first);
+
+		// Set ivRef
+		self->cipherKeys.ivRef = makeString(AES_256_IV_LENGTH, *self->arena);
+		deterministicRandom()->randomBytes(mutateString(self->cipherKeys.ivRef), AES_256_IV_LENGTH);
+		return Void();
+	}
+
+	// Returns the number of bytes that have been written to the buffer
+	static int64_t currentBufferSize(EncryptedRangeFileWriter* self) { return self->wPtr - self->buffer.begin(); }
+
+	static int64_t expectedFileSize(EncryptedRangeFileWriter* self) {
+		// Return what has already been written to file plus the size of the current buffer
+		// which indicates how many bytes the file will contain once the buffer is written
+		return self->file->size() + currentBufferSize(self);
+	}
+
+	static void copyToBuffer(EncryptedRangeFileWriter* self, const void* src, size_t size) {
+		std::memcpy(self->wPtr, src, size);
+		self->wPtr += size;
+		ASSERT(currentBufferSize(self) <= self->blockSize);
+	}
+
+	static void appendStringRefWithLenToBuffer(EncryptedRangeFileWriter* self, StringRef* s) {
+		// Append the string length followed by the string to the buffer
+		uint32_t lenBuf = bigEndian32((uint32_t)s->size());
+		copyToBuffer(self, &lenBuf, sizeof(lenBuf));
+		copyToBuffer(self, s->begin(), s->size());
+	}
+
+	static bool isSystemKey(KeyRef key) { return key.size() && key[0] == systemKeys.begin[0]; }
+
+	ACTOR static Future<std::pair<int64_t, TenantName>> getTenantIdNameImpl(
```

Comment:
The routine enables the caller to procure following information: `encryptionDomainId` and `encryptionDomainName`; it would be better to rename it as `getEncryptionDomainDetails`; especially when `default_tenant_id` concept doesn't exists, but, `default_encrypt_domain_id` is a legit concept in the code. 

There is lot of value in maintaining concept separations in code for more clarity. 
Thanks